app/file.go

```diff
@@ -621,6 +621,30 @@ func (a *App) GetFileInfo(fileId string) (*model.FileInfo, *model.AppError) {
 	return result.Data.(*model.FileInfo), nil
 }
 
+func (a *App) GetFileThumbnail(fileId string) ([]byte, *model.AppError) {
+
+	fileInfo, err := a.GetFileInfo(fileId)
+	if err != nil {
+		mlog.Error(
+			fmt.Sprintf("File not found when getting file thumbnails, err=%v", err),
+			mlog.String("file_id", fileId),
+		)
+		return nil, err
+	}
+
+	thumbnailPath := fileInfo.ThumbnailPath
```

Comment:
If the logging is removed, you can just `return a.ReadFile(fileInfo.ThumbnailPath)` to simplify the code.